src/database/migration/file/migration_version_7.ts

```diff
@@ -0,0 +1,45 @@
+import {SQLiteDatabase} from 'react-native-sqlite-storage';
+
+import ChannelList from '../../schema/ChannelListSchema';
+import ChatSchema from '../../schema/ChatSchema';
+import Migration from './migration.types';
+
+const TABLE_NAME = ChatSchema.getTableName();
+
+class MigrationVersion7 implements Migration {
+  up = async (db: SQLiteDatabase): Promise<void> => {
+    const upQueryStep1 = `CREATE TABLE IF NOT EXISTS ${TABLE_NAME}_temp (
+        id UUID PRIMARY KEY NOT NULL,
+        channel_id TEXT NOT NULL,
+        user_id UUID NOT NULL,
+        message TEXT NOT NULL,
+        type TEXT NOT NULL,
+        status TEXT NOT NULL,
+        created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+        updated_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+        raw_json TEXT,
+        FOREIGN KEY (channel_id) REFERENCES ${ChannelList.getTableName()} (id),
+        FOREIGN KEY (user_id) REFERENCES users (id)
+      );`;
+
+    const upQueryStep2 = `INSERT INTO ${TABLE_NAME}_temp SELECT * FROM ${TABLE_NAME};`;
+    const upQueryStep3 = `DROP TABLE ${TABLE_NAME};`;
+    const upQueryStep4 = `ALTER TABLE ${TABLE_NAME}_temp RENAME TO ${TABLE_NAME};`;
+
+    console.log(`===== MIGRATING: ${TABLE_NAME.toLocaleUpperCase()} TABLE =====`);
+    await db.executeSql(upQueryStep1);
+    await db.executeSql(upQueryStep2);
+    await db.executeSql(upQueryStep3);
+    await db.executeSql(upQueryStep4);
+    console.log(`===== DONE MIGRATING: ${TABLE_NAME.toLocaleUpperCase()}  TABLE =====`);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The migration steps are executed sequentially without any error handling. If one of the steps fails, the subsequent steps will still be attempted, which could lead to an inconsistent database state. It would be better to wrap these steps in a transaction and rollback if any step fails.

```diff
+    db.transaction(async tx => {
+      try {
+        console.log(`===== MIGRATING: ${TABLE_NAME.toLocaleUpperCase()} TABLE =====`);
+        await tx.executeSql(upQueryStep1);
+        await tx.executeSql(upQueryStep2);
+        await tx.executeSql(upQueryStep3);
+        await tx.executeSql(upQueryStep4);
+        console.log(`===== DONE MIGRATING: ${TABLE_NAME.toLocaleUpperCase()}  TABLE =====`);
+      } catch (error) {
+        console.error(`Migration failed for ${TABLE_NAME}`, error);
+        throw error;
+      }
+    });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->